src/vs/editor/contrib/dropOrPasteInto/browser/copyPasteContribution.ts

```diff
@@ -26,8 +29,36 @@ registerEditorCommand(new class extends EditorCommand {
 	}
 
 	public override runEditorCommand(_accessor: ServicesAccessor | null, editor: ICodeEditor, _args: any) {
-		CopyPasteController.get(editor)?.changePasteType();
+		return CopyPasteController.get(editor)?.changePasteType();
 	}
 });
 
-registerEditorFeature(DefaultPasteProvidersFeature);
+registerEditorAction(class extends EditorAction {
+	constructor() {
+		super({
+			id: 'editor.action.pasteAs',
+			label: nls.localize('pasteAs', "Paste As..."),
+			alias: 'Paste As...',
+			precondition: undefined,
+			description: {
+				description: `Paste as`,
+				args: [{
+					name: 'args',
+					schema: {
+						type: 'object',
+						properties: {
+							'id': {
+								type: 'string',
```

Comment:
Is there a limited number of paste as ids? Could this be an enum?

Reply:
No these ids are contributed by extensions, for example `myExt.pasteAttachment` or  `myExt.insertVideo`